matrix-sdk-android/src/main/java/org/matrix/android/sdk/internal/session/room/state/DefaultStateService.kt

```diff
@@ -73,45 +67,41 @@ internal class DefaultStateService @AssistedInject constructor(@Assisted private
         return stateEventDataSource.getStateEventsLive(roomId, eventTypes, stateKey)
     }
 
-    override fun sendStateEvent(
+    override suspend fun sendStateEvent(
             eventType: String,
             stateKey: String?,
-            body: JsonDict,
-            callback: MatrixCallback<Unit>
-    ): Cancelable {
-        val params = SendStateTask.Params(
-                roomId = roomId,
-                stateKey = stateKey,
-                eventType = eventType,
-                body = body
-        )
-        return sendStateTask
-                .configureWith(params) {
-                    this.callback = callback
-                }
-                .executeBy(taskExecutor)
+            body: JsonDict
+    ) {
+        withContext(coroutineDispatchers.main) {
```

Comment:
Why changing the context here?

Reply:
Sorry, didn't read `executeBy(taskExecutor)` internals properly.
TaskExecutor.execute ran this on the callback's thread (by transforming it with `toDispatcher()` internally). I'll remove the context change.